sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/FileQueryAsyncResponse.java

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.file.datalake.models;
+
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.rest.ResponseBase;
+import reactor.core.publisher.Flux;
+
+import java.nio.ByteBuffer;
+
+/**
+ * This class contains the response information returned from the server when running a query on a file.
+ */
+public final class FileQueryAsyncResponse extends ResponseBase<FileQueryHeaders, Flux<ByteBuffer>> {
```

Comment:
Shouldn't we just use ResponseBase<> ? this type doesn't seem to do anything new. If we need we can always introduce this in the future without breaking change.
Or is this generated code?

Reply:
This is more just following the pattern of what we return on downloads. FileQueryHeaders is a generated type